src/NuGet.Core/NuGet.Protocol/HttpSource/HttpRetryHandler.cs

```diff
@@ -87,7 +106,14 @@ public class HttpRetryHandler : IHttpRetryHandler
                             requestUri,
                             (int)request.RequestTimeout.TotalMilliseconds);
                         response = await TimeoutUtility.StartWithTimeout(
-                            timeoutToken => request.HttpClient.SendAsync(requestMessage, request.CompletionOption, timeoutToken),
+                            async timeoutToken =>
+                            {
+                                stopwatch.Start();
+                                headerStopwatch.Start();
+                                var responseMessage = await request.HttpClient.SendAsync(requestMessage, request.CompletionOption, timeoutToken);
```

Comment:
will `request.CompletionOption` always be `ResponseHeadersRead`? If not then it looks like the `stopwatch` vs. `headerStopwatch` distinction is not correct.

Reply:
the default completion option is set here: https://github.com/NuGet/NuGet.Client/blob/3803820961f4d61c06d07b179dab1d0439ec0d91/src/NuGet.Core/NuGet.Protocol/HttpSource/HttpRetryHandlerRequest.cs#L23

It's never assigned any different vault in NuGet's source, so I'd love to remove that flexibility, but it's possible someone uses the `NuGet.Protocol` package and sets it themselves.

Reply:
Might be worth noting in the event then so you have this context in the future if this assumption changes.

Reply:
I made changes to support the scenario.